samples/snippets/quickstart.py

```diff
@@ -0,0 +1,275 @@
+
+    print(f"Deleting {machine_name} from {zone}...")
+    operation = instance_client.delete(
+        project=project_id, zone=zone, instance=machine_name
+    )
+    if operation.status == compute_v1.Operation.Status.RUNNING:
+        operation_client = compute_v1.ZoneOperationsClient()
+        operation = operation_client.wait(
+            operation=operation.name, zone=zone, project=project_id
+        )
+    if operation.error:
+        print("Error during deletion:", operation.error, file=sys.stderr)
+    if operation.warnings:
+        print("Warning during deletion:", operation.warnings, file=sys.stderr)
+    print(f"Instance {machine_name} deleted.")
+    return
+
+
+# [END compute_instances_delete]
+
+
+# [START compute_instances_operation_check]
+def wait_for_operation(
+    operation: compute_v1.Operation, project_id: str
+) -> compute_v1.Operation:
+    """
+    This method waits for an operation to be completed. Calling this function
+    will block until the operation is finished.
+
+    Args:
+        operation: The Operation object representing the operation you want to
+            wait on.
+        project_id: ID or number of the project owning the operation.
+
+    Returns:
+        Finished Operation object.
+    """
+    kwargs = {"project": project_id, "operation": operation.name}
+    if operation.zone:
+        client = compute_v1.ZoneOperationsClient()
+        # Operation.zone is a full URL address of a zone, so we need to extract just the name
+        kwargs["zone"] = operation.zone.rsplit("/", maxsplit=1)[1]
+    elif operation.region:
+        client = compute_v1.RegionOperationsClient()
+        # Operation.region is a full URL address of a zone, so we need to extract just the name
+        kwargs["region"] = operation.region.rsplit("/", maxsplit=1)[1]
+    else:
+        client = compute_v1.GlobalOperationsClient()
+    return client.wait(**kwargs)
+
+
+# [END compute_instances_operation_check]
+
+
+def main(project_id: str, zone: str, instance_name: str) -> None:
+
+    create_instance(project_id, zone, instance_name)
+
+    zone_instances = list_instances(project_id, zone)
+    print(f"Instances found in {zone}:", ", ".join(i.name for i in zone_instances))
+
+    all_instances = list_all_instances(project_id)
+    print(f"Instances found in project {project_id}:")
+    for i_zone, instances in all_instances.items():
+        print(f"{i_zone}:", ", ".join(i.name for i in instances))
+
+    delete_instance(project_id, zone, instance_name)
+
+
+if __name__ == "__main__":
+    import uuid
+    import google.auth
+    import google.auth.exceptions
+
+    try:
+        default_project_id = google.auth.default()[1]
+    except google.auth.exceptions.DefaultCredentialsError:
+        print(
+            "Please use `gcloud auth application-default login` "
+            "or set GOOGLE_APPLICATION_CREDENTIALS to use this script."
+        )
+    else:
+        test_instance_name = "i" + uuid.uuid4().hex[:10]
```

Comment:
Why is this called `test_instance_name`?

Reply:
Other note, can you concatenate something like "quickstart" to the instance name, so it's easier to identify?  I ran this on a current project and got a little confused and worried before I recognized which one was the quickstart.

Reply:
Changed to just `instance_name` and made the name start with `quickstart-`